perf/pgbench.py

```diff
@@ -0,0 +1,143 @@
+from avocado.utils.software_manager.manager import SoftwareManager
+
+
+class Pgbench(Test):
+
+    """
+    This module will run the pgbench benchmark. Pgbench is a benchmark for
+    measuring the throughput achieved by executing database transactions on
+    PostgresSQL database.
+    """
+    sm = SoftwareManager()
+    scaling_factor = worker_threads = db_clients = protocol = \
+        benchmark_duration = transaction_count = 0
+    already_installed = True
+
+    def run_cmd(self, cmd, ignore_failure):
+        command = process.SubProcess(cmd)
+        command.run()
+
+        if command.result.exit_status and ignore_failure:
+            self.log.info(f"Command '{cmd}' returned '{command.result.exit_status}' "
+                          f"with the text '{command.result.stderr_text}'")
+            return [command.result.exit_status, command.result.stderr_text]
+        elif command.result.exit_status:
+            self.cancel(f"Command '{cmd}' returned '{command.result.exit_status}' "
+                        f"with the text '{command.result.stderr_text}'")
+        else:
+            return [command.result.exit_status, command.result.stdout_text]
+
+    def get_pkg_manager(self):
+        for pkg_mgr in ['apt', 'dnf', 'yum']:
+            if self.sm.check_installed(pkg_mgr):
+                return pkg_mgr
+        return None
+
+    def install_and_initialize_postgres(self, pkg_mgr):
+        packages = ["postgresql", "postgresql-contrib"]
+        for package in packages:
+            self.sm.install(package)
+
+        # rhel needs an extra pkg and initialization
+        if pkg_mgr in ['dnf', 'yum']:
+            self.sm.install('postgresql-server')
+            output = self.run_cmd("postgresql-setup --initdb", True)
+            if output[0] and "not empty" not in output[1]:
+                self.cancel(f"Failed to initialize postgres - {output[1]}")
+
+    def setUp(self):
+        '''
+        Install Pgbench
+        '''
+        # check root privileges ($USER should be able to run commands
+        # as another user without requiring a password)
+        if os.geteuid() != 0:
+            self.cancel("This script requires root privileges, "
+                        "Please try again with 'sudo' or as 'root'")
+
+        # check if pgbench is installed
+        if (self.sm.check_installed("pgbench")):
+            self.log.info("Pgbench is already installed")
+        else:
+            pkg_mgr = self.get_pkg_manager()
+            if pkg_mgr:
+                self.install_and_initialize_postgres(pkg_mgr)
+                self.already_installed = False
+            else:
+                self.cancel("Unsupported Linux distribution")
+
+        # restart postgres service
+        ManagerService = ServiceManager()
+        ManagerService.restart("postgresql")
+
+        # add current user to db
+        output = self.run_cmd(f"sudo -u postgres createuser --superuser {os.getlogin()}", True)
```

Comment:
process run has an argument as run as `sudo=True` that will help to run any command on root privileged user

Reply:
Please not that this command has to be run as postgres user. So if we just do `sudo=True` it won't work